spec/system/candidate_interface/preferences/candidate_adds_preferences_spec.rb

```diff
@@ -93,7 +95,40 @@
     then_i_am_redirected_to_review_page_without_locations
 
     when_i_click('Submit preferences')
-    then_i_am_redirected_application_choices_with_success_message
+    then_i_am_redirected_to_application_choices_with_success_message
+  end
+
+  scenario 'Candidate edits a dynamic location' do
+    given_i_am_signed_in
+    given_courses_exist
+    and_feature_flag_is_enabled
+    given_i_am_on_the_share_details_page
+
+    when_i_click('Change your sharing and location settings')
+    then_i_am_redirected_to_opt_in_page
+
+    when_i_opt_in_to_find_a_candidate
+    and_i_click('Continue')
+    then_i_am_redirected_to_location_preferences(location_preferences)
+
+    when_i_check_dynamic_locations
+    and_i_click('Continue')
+    and_i_click('Submit preferences')
+    then_i_am_redirected_to_application_choices_with_success_message
+
+    when_i_click('Add application')
+    and_i_complete_the_flow_for_adding_a_choice
+    then_i_am_redirected_to_application_choices
+
+    when_i_click('Change your sharing and location settings')
+    and_i_click_the_relevant_change_link
+    then_i_see_my_location_preferences_page_including_the_dynamic_location
+
+    when_i_click_on_the_dynamic_location_change_link
+    and_i_change_the_distance
+    and_i_click('Update location')
+    then_i_see_my_location_preferences_page
+    and_the_distance_is_updated
```

Comment:
I would like to see the test more explicitly describe the problem we are solving with this PR. 
eg (this isn't perfect, but 
```
given_i_am_a_candidate_who_has_opted_in_with_dyncamic_location, # This would include creating the submitted application, the feature flag turning on, the opting in with dynamic locations true, etc
and_i_have_a_location_preference_with_a_bogus_name_from_a_dynamic_location
and_i_navigate_to_update_my_preferences
when_i_update_the_radius_only
it_saves_successfully
```
It feels like nearly everything in lines 103 - 121 is background, which makes it difficult to understand exactly what we are testing. It would be hard for me to answer the question 'is there a test for just updating the radius'? if I was looking at this file.